server/protocols/jmap/src/main/java/org/apache/james/jmap/model/JmapMDN.java

```diff
@@ -150,7 +151,7 @@ public Message generateMDNMessage(Message originalMessage, MailboxSession mailbo
             .setTo(getSenderAddress(originalMessage))
             .setFrom(user.asString())
             .setSubject(subject)
-            .setMessageId(MimeUtil.createUniqueMessageId(user.getDomainPart().orElse(null)))
+            .setMessageId(MimeUtil.createUniqueMessageId(user.getDomainPart().map(Domain::name).orElse(null)))
```

Comment:
asString?

Reply:
using the case you have as an input seems a better idea